types.go

```diff
@@ -118,6 +118,32 @@ func (log *Log) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// FilterParams - Filter parameters object
+type FilterParams struct {
```

Comment:
Address must be []string and Topics must be [][]string.
Also you don't need to implement MarshalJSON method, json tags with omitempty will be enough.